src/wallet/rpcwallet.cpp

```diff
@@ -2638,16 +2660,84 @@ static UniValue loadwallet(const JSONRPCRequest& request)
     return obj;
 }
 
+static UniValue setwalletflag(const JSONRPCRequest& request)
+{
+    std::shared_ptr<CWallet> const wallet = GetWalletForJSONRPCRequest(request);
+    CWallet* const pwallet = wallet.get();
+
+    if (!EnsureWalletIsAvailable(pwallet, request.fHelp)) {
+        return NullUniValue;
+    }
+
+    if (request.fHelp || request.params.size() < 1 || request.params.size() > 2) {
+        std::string flags = "";
+        for (auto& it : WALLET_FLAG_MAP)
+            if (it.second & MUTABLE_WALLET_FLAGS)
+                flags += (flags == "" ? "" : ", ") + it.first;
+        throw std::runtime_error(
+            RPCHelpMan{"setwalletflag",
+                "\nChange the state of the given wallet flag for a wallet.\n",
+                {
+                    {"flag", RPCArg::Type::STR, RPCArg::Optional::NO, "The name of the flag to change. Current available flags: " + flags},
+                    {"state", RPCArg::Type::BOOL, /* default */ "true", "The new state."},
+                },
+                RPCResult{
+            "{\n"
+            "    \"success\": true|false (boolean) Boolean value indicating whether the operation was successful or not\n"
+            "    \"error\":   string     (string) Error message, or unset if successful\n"
+            "    \"caveats\": string     (string) Any caveats associated with the change\n"
+            "}\n"
+                },
+                RPCExamples{
+                    HelpExampleCli("setwalletflag", "avoid_reuse")
+                  + HelpExampleRpc("setwalletflag", "\"avoid_reuse\"")
+                },
+            }.ToString());
+    }
+
+    std::string flag_str = request.params[0].get_str();
+    bool value = request.params[1].isNull() || request.params[1].get_bool();
+
+    if (!WALLET_FLAG_MAP.count(flag_str)) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("Unknown wallet flag: %s", flag_str));
+    }
+
+    auto flag = WALLET_FLAG_MAP.at(flag_str);
+
+    if (!(flag & MUTABLE_WALLET_FLAGS)) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("Wallet flag is immutable: %s", flag_str));
+    }
+
+    UniValue res(UniValue::VOBJ);
+
+    if (pwallet->IsWalletFlagSet(flag) == value) {
+        res.pushKV("success", false);
+        res.pushKV("error", strprintf("flag %s set: %s", flag ? "already" : "not", flag_str));
+        return res;
+    }
+
+    res.pushKV("success", true);
+
+    pwallet->SetWalletFlag(flag, value);
+
+    if (flag && WALLET_FLAG_CAVEATS.count(flag)) {
+        res.pushKV("caveats", WALLET_FLAG_CAVEATS.at(flag));
```

Comment:
I like `success` better than simply saying what the flag was changed to, because it better addresses the case where the user is targeting the wrong wallet. For example, I may ssh into a server with a node on it, and my intention is to turn a wallet flag on, but in this scenario, I ssh'd into the wrong machine and/or picked the wrong wallet (in the case of a node with multiple wallets).

With `success=<flag was actually changed>`, I would get a `success=false` here, which would hopefully lead me to discover my error. With `new_flag_value=true` here, I would get no such indication.

Perhaps both would be a good compromise here? Add `flag_name`, `flag_old_value`, `flag_new_value` to the results. Too verbose?

A caveat could apply to turning off a flag as well as turning it on, though, but as this is not the case right now, I'm going with "only show when turning flag to ON" approach.

As a non-native speaker, I don't think caveat is that weird a word, but will change it if people think it's hard to understand.

Reply:
> Add flag_name, flag_old_value, flag_new_value to the results. Too verbose?

Seems ok to me. Alternatively, you could just throw a `JSONRPCError` if the flag is already set to what was requested. That's the way we usually deal with input errors in the RPCs.

> As a non-native speaker, I don't think caveat is that weird a word

I'm going to claim that 'warning' is a [far more common word](https://books.google.com/ngrams/graph?content=caveat%2Cwarning&year_start=1800&year_end=2008&corpus=15&smoothing=7&share=&direct_url=t1%3B%2Ccaveat%3B%2Cc0%3B.t1%3B%2Cwarning%3B%2Cc0), and conveys the same meaning.

Reply:
Throwing seems sensible. Going with that and adding flag name and value (but skipping old value as it's now implied). Switching caveat to warning.